storage/health.go

```diff
@@ -0,0 +1,29 @@
+package storage
+
+import (
+	"fmt"
+	"time"
+)
+
+// NewCustomHealthCheckFunc returns a new health check function.
+func NewCustomHealthCheckFunc(s Storage, now func() time.Time) func() (details interface{}, err error) {
```

Comment:
So if I'm correct, the health check creates and deletes an auth request every time it hits the check endpoint?

Reply:
It does. It's not optimal IMO, but this PR is about moving the health check to the telemetry endpoint.

Most storage solutions implement some sort of Ping under the hood, so we should probably use that in the future.

Reply:
Sorry, not quite. The health check runs in the background with a predefined interval, the health endpoint just reports back the last result. There is no behavior change on this front.

But creating dummy auth requests is probably not a good idea. I'll try to address that in a followup PR, but I need this change for the helm chart.

Reply:
No problem, I'm okay with this right now.